modules/custom/az_paragraphs/src/Plugin/Field/FieldFormatter/AZBackgroundMediaFormatter.php

```diff
@@ -0,0 +1,598 @@
+      '#type' => 'textfield',
+      '#title' => $this->t('Selector'),
+      '#description' => $this->t(
+        'A valid CSS selector that will be used to apply the background image. Tokens are supported.'
+      ),
+      '#default_value' => $settings['css_settings']['selector'],
+    ];
+    // The token help relevant to this entity type.
+    if (isset($form['#entity_type'])) {
+      $form['css_settings']['token_help'] = [
+        '#theme' => 'token_tree_link',
+        '#token_types' => ['user', $form['#entity_type']],
+      ];
+    }
+    else {
+      $form['css_settings']['token_help'] = [
+        '#theme' => 'token_tree_link',
+      ];
+    }
+
+    return $form;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $settings = $this->getSettings();
+    $options = $this->getResponsiveImageStyles();
+
+    if (isset($options[$settings['image_style']])) {
+      $summary[] = $this->t('URL for image style: @style', ['@style' => $options[$settings['image_style']]]);
+    }
+    else {
+      $summary[] = $this->t('Original image style');
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+
+    $settings = $this->getAllSettings($items);
+
+    $element = [];
+
+    $full_width = '';
+    $marquee_style = $settings['style'];
+    if (!empty($settings['full_width'])) {
+      $full_width = $settings['full_width'];
+    }
+    $media_items = $this->getEntitiesToView($items, $langcode);
+
+    // Early opt-out if the field is empty.
+    if (empty($media_items)) {
+      return $element;
+    }
+
+    $paragraph = $items->getEntity();
+
+    // Prepare token data in bg image CSS selector.
+    $token_data = [
+      'user' => $this->currentUser,
+      $items->getEntity()->getEntityTypeId() => $items->getEntity(),
+    ];
+
+    $settings['css_settings']['selector'] = $this->token->replace(
+      $settings['css_settings']['selector'],
+      $token_data
+    );
+    $settings['css_settings'] = str_replace(['_'], '-', $settings['css_settings']);
+
+    /** @var \Drupal\media\MediaInterface[] $media_items */
+    foreach ($media_items as $delta => $media) {
+      $element['#media_type'] = $media->bundle();
+
+      switch ($media->bundle()) {
+        case 'az_remote_video':
+          $element[$delta] = $this->remoteVideo($settings, $media);
+          break;
+
+        case 'az_image':
+          $element[$delta] = $this->image($settings, $media);
+          break;
+
+        default:
+          return $element;
+      }
+
+    }
+
+    return $element;
+  }
+
+  /**
+   * Get the possible responsive image styles.
+   *
+   * @param bool $withNone
+   *   True to include the 'None' option, false otherwise.
+   *
+   * @return array
+   *   The select options.
+   */
+  protected function getResponsiveImageStyles($withNone = FALSE) {
+    $styles = ResponsiveImageStyle::loadMultiple();
```

Comment:
Shouldn't this be using the injected `$entityTypeManager` service?

```suggestion
    $styles = $this->entityTypeManager->getStorage('responsive_image_style')->loadMultiple();
```
This would alsp allow you to remove the `use Drupal\responsive_image\Entity\ResponsiveImageStyle;` at the top of the file.